lib/pact_broker/client/hal/http_client.rb

```diff
@@ -127,6 +133,18 @@ def verbose?
           verbose || ENV["VERBOSE"] == "true"
         end
 
+        def custom_x509_certificate?
+          ENV['X509_CERT_ENABLED'] == 'true'
+        end
+
+        def x509_cert_file
+          File.read(ENV['X509_CERT_FILE_PATH'])
```

Comment:
As Matt said, we do usually prefix all env vars with `PACT_`. We use `SSL_CERT_FILE` and `SSL_CERT_DIR` because they are pre-existing Unix standards, however. 

There's no such standard for x509 certificates that I can find though, so we need to decide whether to make it match the SSL cert env vars, or the rest of the PACT env vars. The SSL env vars and the x509 env vars are likely to be used together, aren't they, given that this is about doing mutual certificate authentication. Given this is the *client* cert (not the server cert) it might make sense to include that in the name.
I think the most consistent approach would be to call them `X509_CLIENT_CERT_FILE` and `X509_CLIENT_KEY_FILE`.

We don't have command line arguments for the SSL certificates, so I don't think we need to add them for the x509 certs.

Reply:
These are client certificates, that's correct. I agree with including that in the name for clarity.

I don't see anywhere in this code that requires the client to _confirm_ the certificate (authenticity) of the server - so perhaps that's forthcoming, or not required for the use case.

i.e. as it stands, a malicious server could present a valid certificate and if it's in the CLI's CA then there are no client-side checks to ensure the requests are going to the right place.